### PR TITLE
oracle: new connector

### LINKED ISSUE #322: support for oracle db
I've been following some threads in https://github.com/oracle/node-oracledb for the last year or two, and it looks like native binaries are finally available! https://github.com/oracle/node-oracledb/issues/18#issuecomment-351625712. As far as I recall, this was one of the blockers for supporting oracledb in this client.

### PR DESCRIPTION
**Pull request by n-riesco**
* Implemented connector to Oracle based on node-oracledb.

* Implemented a pool of clients to ensure all the request from a
  connector share the same client. For more details, see
  `backend/persistent/datastores/pool.js`.

* Upgraded to electron@2.0 and node@8, so that:
  - both node and electron have ABI 57 (for which node-oracledb provides
    binaries)
  - unable to use electron@1.8 due to issue resulting in a empty window
    at some startups

* Added Dockerfile and npm scripts to build and launch a container with
  an Oracle Express database setup for testing.

* Ensure connector failures are logged.

* Added npm script `test-unit-oracle` for testing. See `CONTRIBUTING.md`
  for the requirements to run these tests.

* Updated documentation in `CONTRIBUTING.md`.

Closes #322

### CODE REVIEW

**Comment by n-riesco** [comment]
TODO:

**Comment by n-riesco** [comment]
[Oracle Instant Client (all)](http://www.oracle.com/technetwork/database/database-technologies/instant-client/downloads/index.html)

[Oracle Instant Client (win64)](http://www.oracle.com/technetwork/topics/winx64soft-089540.html)
[Oracle Instant Client (linux)](http://download.oracle.com/otn/linux/oracle11g/xe/oracle-xe-11.2.0-1.0.x86_64.rpm.zip)
[Oracle Instant Client Downloads for macOS (Intel x86)](http://www.oracle.com/technetwork/topics/intel-macsoft-096467.html)
[]()

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ matrix:
     - os: osx
       osx_image: xcode9.0
       language: node_js
-      node_js: "6"
+      node_js: "8"
```
Comment (shannonlal) [inline .travis.yml]: Just to confirm.  Once this is moved in we will need to use NodeJS 8?  Will earlier versions work?  Not an issue just want to know.

**Comment by shannonlal** [comment]
Looks good.  No major issues.  Just had a couple of quick questions for you on but should not block merging this in💃

**Comment by n-riesco** [comment]
I've tested this branch on windows:
- the web app works (i.e. oracledb works in Node)
- the desktop app doesn't (i.e. oracledb fails when required from Electron)

The issue I have with electron app has already been reported: https://github.com/oracle/node-oracledb/issues/917

Running `.\node_modules\.bin\electron-rebuild.cmd --version=2.0.1 --arch=x64 --only=oracledb` as suggested in https://github.com/oracle/node-oracledb/issues/881#issuecomment-375245656 fails with NAN-related errors (like [these](https://gist.github.com/lealhugui/3020caa8b961618c9bf48aa34f1c437c) reported in https://github.com/oracle/node-oracledb/issues/862#issue-300385516) 

---

Potential leads:

- Master contains commits to be released with v2.3 that seem to address the NAN-related errors:
  - https://github.com/oracle/node-oracledb/commit/13680a6f14578aa5f23ef47dfe0f1251446a291e
  - https://github.com/oracle/node-oracledb/commit/00313f840b32348399aea44a649380d36ab6993c

- Solution to investigate: [create our own npm packages](https://github.com/oracle/node-oracledb/issues/862#issuecomment-370977142)

**Comment by n-riesco** [comment]
https://github.com/plotly/falcon-sql-client/pull/437/commits/3c13700d21ca505d699f2d58d89e2b7f1687dc94 fixes the issue with the windows installer.

I've tested both, the linux and the windows installer.

@tarzzz, could you test the mac installer [here](https://s3.amazonaws.com/falcon-travis-artifacts/plotly/falcon-sql-client/603/603.1/mac-release.zip), please?

To test the connector for oracle, you'll need to:
- Install the free Oracle Instant Client for mac, as described [here](https://github.com/oracle/node-oracledb/blob/master/INSTALL.md#354-install-the-free-oracle-instant-client-basic-zip-file)
- Build the docker image with the oracle database for testing: `yarn docker:oracle:build`
- Launch the docker image: `yarn docker:oracle:start`
- Wait until the docker container prints out `Ready`
- Install falcon using the installer [here](https://s3.amazonaws.com/falcon-travis-artifacts/plotly/falcon-sql-client/603/603.1/mac-release.zip)
- Open Falcon and create an connection to oracle using the sample credentials.

**Comment by tarzzz** [comment]
Getting this while trying to connect:

image

It seems an issue with my Oracle client installation.. !!

**Comment by tarzzz** [comment]
Okay, I was able to make it work by correctly installing clients:

image

Everything looks good to me from testing point of view.. 💃

**Comment by n-riesco** [comment]
@tarzzz I'm planning to update README.md with the instructions for installing Oracle Install Client.
For mac, did you have to do anything else other what is described [here](https://github.com/oracle/node-oracledb/blob/master/INSTALL.md#354-install-the-free-oracle-instant-client-basic-zip-file)?

**Pull request merged**